libcloud/compute/drivers/openstack.py

```
@@ -206,6 +206,55 @@ def list_nodes(self):
         return self._to_nodes(
             self.connection.request('/servers/detail').object)
 
+    def create_volume(self, size, name, location=None, snapshot=None):
```

This doesn't seem to handle snapshots, so it should raise an error if a snapshot is provided.